prog/vm/nexus.rb

```diff
@@ -14,7 +14,7 @@ def self.assemble(public_key, project_id, name: nil, size: "standard-2",
     unix_user: "ubi", location: "hetzner-hel1", boot_image: Config.default_boot_image_name,
     private_subnet_id: nil, nic_id: nil, storage_volumes: nil, boot_disk_index: 0,
     enable_ip4: false, pool_id: nil, arch: "x64", allow_only_ssh: false, swap_size_bytes: nil,
-    distinct_storage_devices: false, force_host_id: nil, exclude_host_ids: [])
+    distinct_storage_devices: false, force_host_id: nil, exclude_host_ids: [], gpu_count: 0)
```

Comment:
would it be feasible to "learn" the gpu count, like cores and network?

Reply:
also, maybe you kinda did something about this implicitly?

```
commit 816d182735463338cf443c527cf456c6df3fd759
Author: Benjamin Satzger <ben@ubicloud.com>
Date:   Tue May 7 15:04:26 2024 +0200

    Learn about PCI devices of a host
    
    This commit adds the ability to HostNexus to learn about PCI devices.
    *For now, we only consider NVIDIA devices for which IOMMU is enabled.*
```

Reply:
yes, we already do learn about gpus on hosts. this is for the vm, how many gpus we request for it, similar to how we request cores, memory, ipv4 addresses etc.

Reply:
Of course, I'm not sure what I was thinking.